### PR TITLE
Fix add audio crashes

### LINKED ISSUE #5509: [2.9 ACRA] crash adding audio to card
https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/9a93b88a-ca83-4870-b023-555ca95c5290

```
java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=1, result=-1, data=Intent { dat=content://com.android.providers.media.documents/document/audio:36799 flg=0x1 }} to activity {com.ichi2.anki/com.ichi2.anki.multimediacard.activity.MultimediaEditFieldActivity}: java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
at android.app.ActivityThread.deliverResults(ActivityThread.java:4491)
at android.app.ActivityThread.handleSendResult(ActivityThread.java:4534)
at android.app.ActivityThread.-wrap20(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1752)
at android.os.Handler.dispatchMessage(Handler.java:105)
at android.os.Looper.loop(Looper.java:164)
at android.app.ActivityThread.main(ActivityThread.java:6942)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
Caused by: java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
at com.ichi2.anki.multimediacard.fields.BasicAudioClipFieldController.onActivityResult(BasicAudioClipFieldController.java:9)
at com.ichi2.anki.multimediacard.activity.MultimediaEditFieldActivity.onActivityResult(MultimediaEditFieldActivity.java:3)
at android.app.Activity.dispatchActivityResult(Activity.java:7556)
at android.app.ActivityThread.deliverResults(ActivityThread.java:4487)
... 9 more
```

### PR DESCRIPTION
**Pull request by mikehardy**
## Purpose / Description

Add audio clips was a highly requested feature, and it worked structurally but failed when confronted with unexpected audio clip names or unexpected content resolver states (like, no results - it always expected one)

Also the error handling and implementation of file copies were non-standard

## Fixes
Fixes #5509 

## Approach

Best to review each commit separately as they are discrete and independently useful

1. generalize the ability to show toasts (so we can inform users of audio clip issues)
1. standardize the content file copy and error handling using existing helper function
1. improve content resolver results handling in case results are empty or contain an audio clip with no file extensions

## How Has This Been Tested?

I used an API29 emulator attached to a google play account so I could use my Google Drive as a source of audio files that would not be resolved to file paths.

I added two audio files, one with extension and one without to my google drive

I verified crashes when attempting to add the audio file without an extension

I verified everything works for both files after all the changes

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by mikehardy** [comment]
Had time to fix another common crash - I'd like to get this in 2.9.4 as well

File: AnkiDroid/src/main/java/com/ichi2/anki/UIUtils.java [hunk 1]
```diff
@@ -97,6 +97,13 @@ public static Snackbar showSnackbar(Activity activity, String mainText, int leng
     }
 
 
+    public static void showToast(Context context, CharSequence text) {
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/UIUtils.java]: Seems unnecessary as we already have `showThemedToast()`

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/UIUtils.java]: ! Of course - missed it in the quick scan of UIUtils, switching that commit to bend toast usage to the existing method (it still shouldn't be duplicated where it was, but then the refactor is "just use existing method" vs "extract method")

**Comment by mikehardy** [comment]
All toast usage bent to re-use the existing function I missed seeing previously - you approved the changes before which I will take as "I like the spirit of the change" but your comment about implementation was spot on. I think addressed that so I'll wait for a bit but in the absence of further discussion / you hitting the merge button, I'll merge this and move on. Thanks as ever for the review

**Comment by timrae** [comment]
If I approve with comments it means addressing the comments are optional and you can merge whenever you like

**Pull request merged**